Content.Server/Chat/Systems/ChatSystem.Emote.cs

```diff
@@ -219,14 +238,20 @@ private void InvokeEmoteEvent(EntityUid uid, EmotePrototype proto)
 ///     Use it to play sound, change sprite or something else.
 /// </summary>
 [ByRefEvent]
-public struct EmoteEvent
+public struct EmoteEvent : IInventoryRelayEvent
 {
     public bool Handled;
+    public readonly EntityUid Source;
     public readonly EmotePrototype Emote;
+    public bool Blocked;
 
-    public EmoteEvent(EmotePrototype emote)
+    public EmoteEvent(EntityUid source, EmotePrototype emote)
     {
+        Source = source;
         Emote = emote;
         Handled = false;
+        Blocked = false;
```

Comment:
> You could also probably get away with just setting handled=true to "cancel" it but I personally think that's a little weird.

I think that'd actually be incorrect in this case since the "handled" really only refers to "playing the audio was handled", rather than the entire emote including chat messages.

Reply:
> Perhaps I'm missing something as I'm not that familiar with C#, but both CancellableEntityEventArgs and HandledEntityEventArgs are classes, so EmoteEvent cannot extend both.
Is there some other way to do this (without changing Toolbox or something)?

Hmm yeah your right... I swore there was away to do this but maybe I'm losing my mind 😆
Anyway, I don't think it matters too much so you can probably leave it as is. A maintainer will probably know the real answer!